sdk/tables/azure-data-tables/tests/test_table_client.py

```diff
@@ -253,8 +252,76 @@ def test_client_with_sas_token(self, tables_storage_account_name, tables_primary
             )
             for e in entities:
                 pass
+
+        with pytest.raises(ValueError) as ex:
+            client = TableClient.from_table_url(sas_url, credential=AzureSasCredential(sas_token))
+        ex_msg = "You cannot use AzureSasCredential when the resource URI also contains a Shared Access Signature."
+        assert ex_msg == str(ex.value)
+
+        with TableClient.from_table_url(sas_url, credential=self.get_token_credential()) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        with TableClient.from_table_url(sas_url, credential=tables_primary_storage_account_key) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
             client.delete_table()
 
+    @tables_decorator
+    @recorded_by_proxy
+    def test_client_with_token_credential(self, tables_storage_account_name, tables_primary_storage_account_key):
+        base_url = self.account_url(tables_storage_account_name, "table")
+        table_name = self.get_resource_name("mytable")
+        default_azure_credential = self.get_token_credential()
+
+        with TableServiceClient(base_url, credential=default_azure_credential) as client:
+            client.create_table(table_name)
+            name_filter = "TableName eq '{}'".format(table_name)
+            result = client.query_tables(name_filter)
+            assert len(list(result)) == 1
+
+        with TableClient(base_url, table_name, credential=default_azure_credential) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        with TableClient.from_table_url(f"{base_url}/{table_name}", credential=default_azure_credential) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+    @tables_decorator
+    @recorded_by_proxy
+    def test_client_without_credential(self, tables_storage_account_name, tables_primary_storage_account_key):
+        base_url = self.account_url(tables_storage_account_name, "table")
+        table_name = self.get_resource_name("mytable")
+
+        with pytest.raises(ValueError) as ex:
```

Comment:
This is interesting... as technically this _should_ be supported....
Though I guess raising an error is better than doing something unexpected. We can always add support if we ever get asked to.